src/broadcast.rs

```diff
@@ -89,19 +86,68 @@ pub struct Broadcast<NodeUid> {
     /// Whether we have already output a value.
     has_output: bool,
```

Comment:
Is `has_output` equivalent to `output.is_some()`?

Reply:
No, because the `next_output` method empties `self.output`: It should return `Some` only once, but we need to remember whether we already output something before.